packages/packages/store/src/__tests__/store.test.tsx

```diff
@@ -0,0 +1,271 @@
+	Dispatch,
+	AnyAction,
+} from '../index';
+
+interface SliceStateRoot {
+	slice: {
+		value?: number;
+	}
+}
+
+interface Config {
+	initialValue?: SliceStateRoot['slice']['value'];
+}
+
+const createStoreEntities = ( { initialValue = 1 }: Config = {} ) => {
+	const slice = addSlice( {
+		name: 'slice',
+		initialState: {
+			value: initialValue,
+		},
+		reducers: {
+			setValue: ( state, action ) => {
+				state.value = action.payload;
+			},
+		},
+	} );
+
+	const store = createStore();
+
+	const wrapper: React.FC = ( { children } ) => (
+		<StoreProvider store={ store }>
+			{ children }
+		</StoreProvider>
+	);
+
+	return {
+		slice,
+		store,
+		wrapper,
+	};
+};
+
+describe( '@elementor/store', () => {
+	afterEach( () => {
+		deleteStore();
+	} );
+
+	it( 'should set an initial state of a slice', () => {
+		// Arrange.
+		const { wrapper } = createStoreEntities();
+
+		const { result } = renderHook( () => useSelector( ( state: SliceStateRoot ) => state.slice.value ), { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should update the state value of the slice', () => {
+		// Arrange.
+		const { slice, wrapper } = createStoreEntities();
+
+		// Act.
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 3 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 3 );
+	} );
+
+	it( 'should throw an error when trying to add a slice with the same name more than once', () => {
+		// Arrange.
+		addSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {},
+		} );
+
+		// Assert.
+		expect( () => {
+			addSlice( {
+				name: 'slice',
+				initialState: {
+					value: 1,
+				},
+				reducers: {},
+			} );
+		} ).toThrow( 'Slice with name "slice" already exists.' );
+	} );
+
+	it( 'should throw an error when trying to re-create the store', () => {
+		// Arrange.
+		addSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {},
+		} );
+
+		createStore();
+
+		// Assert.
+		expect( () => createStore() ).toThrow( 'The store instance already exists.' );
+	} );
+
+	it( 'should add a middleware that blocks the state update by not running next(action)', () => {
+		// Arrange.
+		addMiddleware( () => () => () => {} );
+
+		const { slice, wrapper } = createStoreEntities();
+
+		// Act.
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should add a middleware that does not interfere with the state update', () => {
+		// Arrange.
+		addMiddleware( () => ( next: Dispatch<AnyAction> ) => ( action: any ) => {
+			next( action );
+		} );
```

Comment:
maybe add some mock here to make sure it's being executed?

```suggestion
		const a = jest.fn();
		
		addMiddleware( () => ( next: Dispatch<AnyAction> ) => ( action: any ) => {
			a( action );
			next( action );
		} );
		
		expect(a).toHaveBeenCalledOnceWith( ... );
```

Reply:
If the middlewares are not really added/triggered then the previous test will not pass (the test that blocks the state update) do you still think that it's needed?